### PR TITLE
Update last module

### LINKED ISSUE #464: new module: last
Hello,

we would like to submit modules related to the [LAST](https://gitlab.com/mcfrith/last) aligner, which we use for pairwise genome alignment and also mapping of Nanopore reads on genomes.  At the moment we are preparing 7 subtools.  Each of these subtools runs one command that is provided by the `last` package on _Bioconda_, and output their results in a single channel. (Of course the modules also have the channel for version numbers.)

 - [x]  `last/lastdb` for creating a _LAST_ index. (command `lastdb`)
 - [x] `last/train` for training parameters. (command `last-train`)
 - [x] `last/lastal` for the alignment itself. (command `lastal`)
 - [x] `last/split` for selecting optimal one-to-one alignments. (command `last-split`)
 - [x] `last/mafswap` for reversing query and target in the alignment. (command `maf-swap`)
 - [x] `last/mafconvert` for converting the alignment files to various formats such as GFF3. (command `maf-convert`)
 - [x] `last/dotplot` for plotting pairwise alignments. (command `last-dotplot`)

Most subtools have a single input, but the most complex one `last/lastal`, has three input channels: one for the indexed sequences, one for an optional (dummy if needed) file for trained parameters, and one for the sequences to align.

To keep things simple when combining the modules, we decided that all input channels are tuples with `val(meta)` and `path(some input files)`. Sometimes, the `meta` information is just ignored. Please let us know if this is a problem.

The core way to use the modules is: (in pseudo-code)

```
lastdb(target sequences)
train(lastdb.out, query sequences)
lastal(lastdb.out, train.out, query sequences)
```

This pattern allows the user to align genomes to genomes, reads to genomes and even genomes to protein databases.

Because in order to test some of the modules we need to produce their input with other modules, I plan to prepare a single pull request at once. Please let me know if this is a problem.

_edit: turned list into checklist_

### PR DESCRIPTION
**Pull request by charles-plessy**
This is the final PR before closing Issue #464 ! Here are a couple of updates and corrections.

 - Update LAST to version 1238 and MD5 checksums when needed.
 - Update functions.nf to the latest devel version.
 - Make a portable use of `zcat` for MacOS X.
 - Allow input alignments to be uncompressed when possible.
 - Correct how LAST index name was guessed.
 - Use value input channel to chose output file format.
 - Use optional output channels to handle variable file formats. 

Each commit corresponds roughly to one of these items.

## PR checklist

Closes #464

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `<SOFTWARE>.version.txt` file.
- [ ] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Comment by charles-plessy** [comment]
The pytest of `last/dotplot` passes on _singularity_ and _docker_ but fails on _conda_. Maybe we can put the blame on _conda_?

File: software/last/mafconvert/main.nf [hunk 1]
```diff
@@ -11,26 +11,34 @@ process LAST_MAFCONVERT {
...
-        container "https://depot.galaxyproject.org/singularity/last:1219--h2e03b76_0"
+        container "https://depot.galaxyproject.org/singularity/last:1238--h2e03b76_0"
     } else {
-        container "quay.io/biocontainers/last:1219--h2e03b76_0"
+        container "quay.io/biocontainers/last:1238--h2e03b76_0"
     }
 
     input:
     tuple val(meta), path(maf)
+    val(format)
```
Comment (drpatelh) [inline software/last/mafconvert/main.nf:23]:
```suggestion
    val format
```

**Review by drpatelh — APPROVED** [review]
Minor syntactical comment that I would have pushed if I had write permissions to the `luslab` repo! Otherwise looks ab fab!

File: software/last/mafconvert/main.nf (hunk 1 above)
Comment (charles-plessy) [reply software/last/mafconvert/main.nf:23]: Thanks !  I will try to remember fixing that next time I update the software version.  Or in the meantime of course anybody please feel free to make that change.

**Pull request merged**